test/server_error.js

```diff
@@ -1,47 +1,72 @@
-var steed = require("steed");
+var steed = require('steed');
 
-var moscaSettings = function() {
-    return {
-        port: 1883
-    };
+var moscaSettings = function () {
+  return {
+    port: 1883,
+    http: {
+      port: 8000
+    }
+  };
 };
 
-describe("mosca.Server.error", function() {
-    var instance;
-    var secondInstance;
-    var settings;
+var moscaSettings2 = function () {
+  return {
+    port: 1884,
+    http: {
+      port: 8000
+    }
+  };
+};
 
-    beforeEach(function(done) {
-        settings = moscaSettings();
-        settings.publishNewClient = false;
-        settings.publishClientDisconnect = false;
-        instance = new mosca.Server(settings, done);
-        this.instance = instance;
-        this.settings = settings;
-        secondInstance = null;
 
-    });
+describe('mosca.Server.error', function () {
+  var instance;
+  var secondInstance;
 
-    afterEach(function(done) {
-        var instances = [this.instance];
+  beforeEach(function (done) {
+    instance = null;
+    secondInstance = null;
+    done();
+  });
 
-        if (secondInstance) {
-            instances.push(secondInstance);
-        }
+  afterEach(function (done) {
+    this.instance = instance;
+    var instances = [this.instance];
 
-        steed.each(instances, function(instance, cb) {
-            instance.close(cb);
-        }, function() {
-            setImmediate(done);
-        });
+    if (secondInstance) {
+      instances.push(secondInstance);
+    }
+
+    steed.each(instances, function (instance, cb) {
+      instance.close(cb);
+    }, function () {
+      setImmediate(done);
+    });
+  });
+  it('should get MQTT port Error: listen EADDRINUSE', function (done) {
+    this.timeout(10000); // have to wait for the inject with delay of two seconds
+    instance = new mosca.Server(moscaSettings(), function (err, server) {
+      expect(server === instance).to.be.true;
+    });
+    secondInstance = new mosca.Server(moscaSettings(), function (err, server) {
+      expect(server === secondInstance).to.be.true;
+    });
+    secondInstance.on('error', function (err) {
+      expect(err.toString().substr(0, 24)).to.be.equal('Error: listen EADDRINUSE');
+      done();
+    });
+  });
+  it('should get HTTP port Error: listen EADDRINUSE', function (done) {
+    this.timeout(10000); // have to wait for the inject with delay of two seconds
+    instance = new mosca.Server(moscaSettings(), function (err, server) {
```

Comment:
can you please add an assertion that checks that `err` is null or undefined? Also in all other cases.